File: src/main/java/com/github/Doomsdayrs/Jikan4java/types/Support/Stats/Stats.java
package com.github.Doomsdayrs.Jikan4java.types.Support.Stats;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.github.Doomsdayrs.Jikan4java.types.Support.Stats.Score.Score;

/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 * ====================================================================
 * Jikan4java
 * 29 / October / 2018
 *
 * @author github.com/doomsdayrs
 */
public class Stats {
    @JsonProperty("request_hash")
    public String request_hash;

    @JsonProperty("request_cached")
    public boolean request_cached;

    @JsonProperty("request_cache_expiry")
    public int request_cache_expiry;

    @JsonProperty("watching")
    public int watching;

    @JsonProperty("completed")
    public int completed;

    @JsonProperty("on_hold")
    public int on_hold;

    @JsonProperty("dropped")
    public int dropped;

    @JsonProperty("plan_to_watch")
    public int plan_to_watch;

    @JsonProperty("total")
    public int total;

    @JsonProperty("scores")
    public Score scores;
    
    @Override
    public String toString() {
        return "Stats{" +
                "request_hash='" + request_hash + '\'' +
                ", request_cached=" + request_cached +
                ", request_cache_expiry=" + request_cache_expiry +
                ", watching=" + watching +
                ", completed=" + completed +
                ", on_hold=" + on_hold +
                ", dropped=" + dropped +
                ", plan_to_watch=" + plan_to_watch +
                ", total=" + total +
                ", scores=" + scores.toString() +
                '}';
    }
}

File: src/main/java/com/github/doomsdayrs/jikan4java/data/model/support/stats/Stats.kt
package com.github.doomsdayrs.jikan4java.data.model.support.stats

import com.github.doomsdayrs.jikan4java.data.model.support.RequestHashing
import com.github.doomsdayrs.jikan4java.data.model.support.stats.score.Score

/*
 * This file is part of Jikan4java.
 *
 * Jikan4java is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Jikan4java is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Jikan4java.  If not, see <https://www.gnu.org/licenses/>.
 * ====================================================================
 */
/**
 * Jikan4java
 * 29 / October / 2018
 *
 * @author github.com/doomsdayrs
 */
interface Stats : RequestHashing {
	val completed: Int
	val onHold: Int

	@Suppress("PropertyName")
	@Deprecated("Changed to proper naming format", ReplaceWith("onHold"))
	val on_hold: Int
		get() = onHold

	val dropped: Int
	val total: Int
	val scores: Score
}